Improve readme + Update bc.yml and code of conduct (#574)
yiisoft/validator: README.md

composer require yiisoft/validator
```

## General usage

> Validator allows to check data in any format. For example, object:

Reviewer: ```suggestion
Validator allows to check data in any format. For example, when data is an object:
```

```php
use Yiisoft\Validator\Rule\AtLeast;
use Yiisoft\Validator\Rule\Email;
use Yiisoft\Validator\Rule\Length;
use Yiisoft\Validator\Rule\Number;
use Yiisoft\Validator\Rule\Required;
use Yiisoft\Validator\Validator;

#[AtLeast(['email', 'phone'])]
final class Person
{
    public function __construct(
        #[Required]
        #[Length(min: 2)]
        public ?string $name = null,

        #[Number(min: 21)]
        public ?int $age = null,

        #[Email]
        public ?string $email = null,

        public ?string $phone = null,
    ) {
    }
}

$person = new Person(
    name: 'John', 
    age: 17, 
    email: 'john@example.com',
    phone: null
);

$result = (new Validator())->validate($person);
```

> The validation result is an object that allow check validation is successful:

Reviewer: ```suggestion
The validation result is an object that allows to check whether validation was successful:
```

```php
$result->isValid();
```

> …and contain of errors occurred during validation:

Reviewer: ```suggestion
It also contains errors occurred during validation:
```
